MaterialLib/SolidModels/ViscoElasticBurgers-impl.h

```diff
@@ -0,0 +1,332 @@
+                 (K_loc * inc_loc + res_loc).norm());
+        }
+        {
+            Eigen::Matrix<double, KelvinVectorSize * 3, 1> inc_loc =
+                K_loc.householderQr().solve(-res_loc);
+            DBUG("Local Newton: linear solver accuracy |J dx - r| = %g",
+                (K_loc * inc_loc + res_loc).norm());
+        }
+        {
+            Eigen::Matrix<double, KelvinVectorSize * 3, 1> inc_loc =
+                K_loc.colPivHouseholderQr().solve(res_loc);
+            DBUG("colPivHouseHQr linear solver accuracy |J dx - r| = %g",
+                 (K_loc * inc_loc + res_loc).norm());
+        }
+        {
+            Eigen::Matrix<double, KelvinVectorSize * 3, 1> inc_loc =
+                K_loc.fullPivHouseholderQr().solve(-res_loc);
+            DBUG("fullPivHouseHQR linear solver accuracy |J dx - r| = %g",
+                 (K_loc * inc_loc + res_loc).norm());
+        }
+        {
+            Eigen::Matrix<double, KelvinVectorSize * 3, 1> inc_loc =
+                K_loc.llt().solve(-res_loc);
+            DBUG("llt linear solver accuracy |J dx - r| = %g",
+                 (K_loc * inc_loc + res_loc).norm());
+        }
+        {
+            Eigen::Matrix<double, KelvinVectorSize * 3, 1> inc_loc =
+                K_loc.ldlt().solve(-res_loc);
+            DBUG("ldlt linear solver accuracy |J dx - r| = %g",
+                 (K_loc * inc_loc + res_loc).norm());
+        }
+        */
+        Eigen::Matrix<double, KelvinVectorSize * 3, 1> inc_loc =
+            linear_solver.compute(K_loc).solve(-res_loc);
+        DBUG("Local linear solver accuracy |J dx - r| = %g",
+             (K_loc * inc_loc + res_loc).norm());
+
+        // increment solution vectors
+        sigd_j.noalias() += inc_loc.template block<KelvinVectorSize, 1>(
+            KelvinVectorSize * 0, 0);
+        _state._eps_K_j.noalias() +=
+            inc_loc.template block<KelvinVectorSize, 1>(KelvinVectorSize * 1,
+                                                        0);
+        _state._eps_M_j.noalias() +=
+            inc_loc.template block<KelvinVectorSize, 1>(KelvinVectorSize * 2,
+                                                        0);
+
+        // Calculate effective stress and update material properties
+        sig_eff = MaterialLib::SolidModels::Invariants<
+            KelvinVectorSize>::equivalentStress(sigd_j);
+        UpdateBurgersProperties(t, x, sig_eff * _state.GM, _state);
+
+        // evaluation of new residual
+        CalResidualBurgers(dt, epsd_i, sigd_j, _state._eps_K_j, _state._eps_K_t,
+                           _state._eps_M_j, _state._eps_M_t, res_loc, _state);
+        DBUG("Local Newton: Iteration #%d |dx| = %g, |r| = %g", counter,
+             inc_loc.norm(), res_loc.norm());
+    }
+
+    // Hydrostatic part for the stress and the tangent.
+    double const eps_i_trace =
+        MaterialLib::SolidModels::Invariants<KelvinVectorSize>::trace(eps);
+
+    sigma =
```

Comment:
I'm surprised, that even in the case of assignment (not an update like +=) the noalias() made the code shorter.